python/lsst/daf/butler/cli/cmd/commands.py

```diff
@@ -539,3 +541,55 @@ def collection_chain(**kwargs):
     """
     chain = script.collectionChain(**kwargs)
     print(f"[{', '.join(chain)}]")
+
+
+@click.command(cls=ButlerCommand)
+@repo_argument(required=True)
+@click.argument("dataset_type", required=True)
+@click.argument("run", required=True)
+@click.argument("table_file", required=True)
+@click.option("--formatter", type=str,
+              help="Fully-qualified python class to use as the Formatter. If not specified the formatter"
+              " will be determined from the dataset type and datastore configuration.")
+@click.option("--id-generation-mode",
+              default="UNIQUE",
+              help="Mode to use for generating dataset IDs. The default creates a unique ID. Other options"
+              " are: 'DATAID_TYPE' for creating a reproducible ID from the dataID;"
+              " 'DATAID_TYPE_RUN' for creating a reproducible ID from the dataID and run.  The latter is"
+              " usually used for 'raw'-type data.",
```

Comment:
If you want a particular ingested file to have the same UUID wherever you ingest it you must change this mode accordingly. For "raw" in `butler ingest-raws` we use DATAID_TYPE_RUN so that we guarantee that when a raw is ingested into the default collection you end up with the same UUID in every site (so summit, base, IDF, NCSA) -- this helps a lot with provenance. I can't tell with this command whether people want that reproducible UUID so I have to give people the option. Maybe the default should change to DATAID_TYPE_RUN (which means the UUID is derived from the dataId, the datasetType, and the run) but UNIQUE is probably fine as a default as well. So I think if you are running this command you can ignore it if you don't want reproducibility across multiple repos but I have to leave the option here for the cases where people do want it.